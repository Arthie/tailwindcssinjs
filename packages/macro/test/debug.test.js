const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester.default({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import tw from '../lib/macro'`,
    "correct usage 1": `
      import tw from '../lib/macro';
      const css = tw("sm:motion-reduce:translate-y-0 motion-reduce:hover:translate-y-0 md:motion-reduce:hover:translate-y-0");
      `,
  },
});
