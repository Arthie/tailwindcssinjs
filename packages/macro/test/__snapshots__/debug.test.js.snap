// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage 1: correct usage 1 1`] = `

import tw from '../lib/macro';
const css = tw("sm:motion-reduce:translate-y-0 motion-reduce:hover:translate-y-0 md:motion-reduce:hover:translate-y-0");

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "@media (min-width: 640px)": {
    "@media (prefers-reduced-motion: reduce)": {
      "--transform-translate-y": "0",
    },
  },
  "@media (prefers-reduced-motion: reduce)": {
    "&:hover": {
      "--transform-translate-y": "0",
    },
  },
  "@media (min-width: 1024px)": {
    "@media (prefers-reduced-motion: reduce)": {
      "&:hover": {
        "--transform-translate-y": "0",
      },
    },
  },
};


`;

exports[`macros no usage: no usage 1`] = `

import tw from '../lib/macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;
